### PR TITLE
Add xml generator

### PR DESCRIPTION
**Pull request by dkosowski87**
## Changes
Added XMLBuilder dependency.
Added filterSet XML generation functionality.

### CODE REVIEW

File: test/buildFilterSetXML.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const expect = require('chai').expect;
+const buildFilterSetXML = require('../lib/buildFilterSetXML.js');
```
Comment (RafalWilinski) [inline test/buildFilterSetXML.js]: Skip `.js`

File: test/buildFilterSetXML.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    <apps:property name="to" value="(test2@example.com AND test3@example.com)"/>
+    <apps:property name="subject" value="-{Test}"/>
+    <apps:property name="doesNotHaveTheWord" value="test"/>
+    <apps:property name="hasAttachment" value="true"/>
+    <apps:property name="shouldStar" value="true"/>
+    <apps:property name="shouldNeverSpam" value="true"/>
+  </entry>
+</feed>`;
+
+  it('returns a valid xml containing the filter properties', () => {
```
Comment (netguru-hound) [inline test/buildFilterSetXML.js]: 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: test/buildFilterSetXML.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    to: '(test2@example.com AND test3@example.com)',
+    subject: '-{Test}',
+    doesNotHaveTheWord: 'test',
+    hasAttachment: 'true',
+    shouldStar: 'true',
+    shouldNeverSpam: 'true',
+  };
+
+  const properXML = `<?xml version="1.0" encoding="UTF-8"?>
+<feed xmlns="http://www.w3.org/2005/Atom" xmlns:apps="http://schemas.google.com/apps/2006">
```
Comment (netguru-hound) [inline test/buildFilterSetXML.js]: Line is too long.

File: test/buildFilterSetXML.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    from: 'test1@example.com',
+    to: '(test2@example.com AND test3@example.com)',
+    subject: '-{Test}',
+    doesNotHaveTheWord: 'test',
+    hasAttachment: 'true',
+    shouldStar: 'true',
+    shouldNeverSpam: 'true',
+  };
+
+  const properXML = `<?xml version="1.0" encoding="UTF-8"?>
```
Comment (netguru-hound) [inline test/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'template literal syntax' is only available in ES6 (use 'esversion: 6').

File: test/buildFilterSetXML.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    from: '(test1@example.com OR test2example.com)',
+    to: 'test3@example.com',
+    subject: '"This is a test"',
+    hasTheWord: 'test',
+    label: 'Test',
+    shouldArchive: 'true',
+    shouldMarkAsRead: 'true',
+  };
+
+  const filterAttrTwo = {
```
Comment (netguru-hound) [inline test/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: test/buildFilterSetXML.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const expect = require('chai').expect;
+const buildFilterSetXML = require('../lib/buildFilterSetXML');
+
+describe('buildFilterSetXML()', () => {
+  const filterAttrOne = {
```
Comment (netguru-hound) [inline test/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: test/buildFilterSetXML.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const expect = require('chai').expect;
+const buildFilterSetXML = require('../lib/buildFilterSetXML');
+
+describe('buildFilterSetXML()', () => {
```
Comment (netguru-hound) [inline test/buildFilterSetXML.js]: 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: test/buildFilterSetXML.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const expect = require('chai').expect;
+const buildFilterSetXML = require('../lib/buildFilterSetXML');
```
Comment (netguru-hound) [inline test/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

File: test/buildFilterSetXML.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const expect = require('chai').expect;
```
Comment (netguru-hound) [inline test/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>Redefinition of 'expect'.<br>'require' is not defined.

File: test/buildFilterSetXML.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
+'use strict';
```
Comment (netguru-hound) [inline test/buildFilterSetXML.js]: Use the function form of "use strict".

File: lib/buildFilterSetXML.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    const entryNode = feed.element('entry');
+    createStaticFilterNodes(entryNode);
+    createFilterPropertyNodes(entryNode, filter);
+  })
+);
+
+const generateXMLString = feed => feed.end({ pretty: true });
+
+const buildFilterSetXML = (filters) => {
+  const feed = createFeedTemplate();
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: lib/buildFilterSetXML.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+  filters.forEach((filter) => {
+    const entryNode = feed.element('entry');
+    createStaticFilterNodes(entryNode);
+    createFilterPropertyNodes(entryNode, filter);
+  })
+);
+
+const generateXMLString = feed => feed.end({ pretty: true });
+
+const buildFilterSetXML = (filters) => {
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: lib/buildFilterSetXML.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+const createFilterNodes = (feed, filters) => (
+  filters.forEach((filter) => {
+    const entryNode = feed.element('entry');
+    createStaticFilterNodes(entryNode);
+    createFilterPropertyNodes(entryNode, filter);
+  })
+);
+
+const generateXMLString = feed => feed.end({ pretty: true });
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: lib/buildFilterSetXML.js [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+const createFilterPropertyNodes = (entryNode, filter) => (
+  Object.keys(filter).forEach(propertyName => (
+    createFilterPropertyNode(entryNode, propertyName, filter[propertyName])
+  ))
+);
+
+const createFilterNodes = (feed, filters) => (
+  filters.forEach((filter) => {
+    const entryNode = feed.element('entry');
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: lib/buildFilterSetXML.js [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+);
+
+const createFilterPropertyNodes = (entryNode, filter) => (
+  Object.keys(filter).forEach(propertyName => (
+    createFilterPropertyNode(entryNode, propertyName, filter[propertyName])
+  ))
+);
+
+const createFilterNodes = (feed, filters) => (
+  filters.forEach((filter) => {
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: lib/buildFilterSetXML.js [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+    .element('title', ELEMENT_TITLE)
+);
+
+const createFilterPropertyNodes = (entryNode, filter) => (
+  Object.keys(filter).forEach(propertyName => (
+    createFilterPropertyNode(entryNode, propertyName, filter[propertyName])
+  ))
+);
+
+const createFilterNodes = (feed, filters) => (
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: lib/buildFilterSetXML.js [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+);
+
+const createStaticFilterNodes = entryNode => (
+  entryNode
+    .element('category', { term: 'filter' }).up()
+    .element('title', ELEMENT_TITLE)
+);
+
+const createFilterPropertyNodes = (entryNode, filter) => (
+  Object.keys(filter).forEach(propertyName => (
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: lib/buildFilterSetXML.js [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+  entryNode.element('apps:property', { name: propertyName, value: propertyValue })
+);
+
+const createStaticFilterNodes = entryNode => (
+  entryNode
+    .element('category', { term: 'filter' }).up()
+    .element('title', ELEMENT_TITLE)
+);
+
+const createFilterPropertyNodes = (entryNode, filter) => (
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: lib/buildFilterSetXML.js [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+    .attribute('xmlns:apps', APP_NAMESPACE)
+    .element('title', MAIN_TITLE)
+    .up()
+);
+
+const createFilterPropertyNode = (entryNode, propertyName, propertyValue) => (
+  entryNode.element('apps:property', { name: propertyName, value: propertyValue })
+);
+
+const createStaticFilterNodes = entryNode => (
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: lib/buildFilterSetXML.js [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+const createFeedTemplate = () => (
+  XMLBuilder.create('feed', { version: '1.0', encoding: 'UTF-8' })
+    .attribute('xmlns', MAIN_NAMESPACE)
+    .attribute('xmlns:apps', APP_NAMESPACE)
+    .element('title', MAIN_TITLE)
+    .up()
+);
+
+const createFilterPropertyNode = (entryNode, propertyName, propertyValue) => (
+  entryNode.element('apps:property', { name: propertyName, value: propertyValue })
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: Line is too long.

File: lib/buildFilterSetXML.js [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+
+const createFeedTemplate = () => (
+  XMLBuilder.create('feed', { version: '1.0', encoding: 'UTF-8' })
+    .attribute('xmlns', MAIN_NAMESPACE)
+    .attribute('xmlns:apps', APP_NAMESPACE)
+    .element('title', MAIN_TITLE)
+    .up()
+);
+
+const createFilterPropertyNode = (entryNode, propertyName, propertyValue) => (
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: lib/buildFilterSetXML.js [hunk 22]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const XMLBuilder = require('xmlbuilder');
+
+const MAIN_NAMESPACE = 'http://www.w3.org/2005/Atom';
+const APP_NAMESPACE = 'http://schemas.google.com/apps/2006';
+const MAIN_TITLE = 'Mail Filters';
+const ELEMENT_TITLE = 'Mail Filter';
+
+const createFeedTemplate = () => (
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: lib/buildFilterSetXML.js [hunk 23]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const XMLBuilder = require('xmlbuilder');
+
+const MAIN_NAMESPACE = 'http://www.w3.org/2005/Atom';
+const APP_NAMESPACE = 'http://schemas.google.com/apps/2006';
+const MAIN_TITLE = 'Mail Filters';
+const ELEMENT_TITLE = 'Mail Filter';
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: lib/buildFilterSetXML.js [hunk 24]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const XMLBuilder = require('xmlbuilder');
+
+const MAIN_NAMESPACE = 'http://www.w3.org/2005/Atom';
+const APP_NAMESPACE = 'http://schemas.google.com/apps/2006';
+const MAIN_TITLE = 'Mail Filters';
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: lib/buildFilterSetXML.js [hunk 25]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const XMLBuilder = require('xmlbuilder');
+
+const MAIN_NAMESPACE = 'http://www.w3.org/2005/Atom';
+const APP_NAMESPACE = 'http://schemas.google.com/apps/2006';
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: lib/buildFilterSetXML.js [hunk 26]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const XMLBuilder = require('xmlbuilder');
+
+const MAIN_NAMESPACE = 'http://www.w3.org/2005/Atom';
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: lib/buildFilterSetXML.js [hunk 27]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const XMLBuilder = require('xmlbuilder');
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

File: lib/buildFilterSetXML.js [hunk 28]
```diff
@@ -0,0 +1,50 @@
+'use strict';
```
Comment (netguru-hound) [inline lib/buildFilterSetXML.js]: Use the function form of "use strict".

File: test/searchOperators.js [hunk 29]
```diff
@@ -1,11 +1,11 @@
...
-const searchORExpression = require('../lib/searchOperators.js').searchORExpression;
-const searchANDExpression = require('../lib/searchOperators.js').searchANDExpression;
-const searchNOTExpression = require('../lib/searchOperators.js').searchNOTExpression;
-const searchPhrase = require('../lib/searchOperators.js').searchPhrase;
-const searchExactMatch = require('../lib/searchOperators.js').searchExactMatch;
+const searchORExpression = require('../lib/searchOperators').searchORExpression;
+const searchANDExpression = require('../lib/searchOperators').searchANDExpression;
+const searchNOTExpression = require('../lib/searchOperators').searchNOTExpression;
+const searchPhrase = require('../lib/searchOperators').searchPhrase;
+const searchExactMatch = require('../lib/searchOperators').searchExactMatch;
```
Comment (netguru-hound) [inline test/searchOperators.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

File: test/searchOperators.js [hunk 30]
```diff
@@ -1,11 +1,11 @@
...
 const expect = require('chai').expect;
-const searchORExpression = require('../lib/searchOperators.js').searchORExpression;
-const searchANDExpression = require('../lib/searchOperators.js').searchANDExpression;
-const searchNOTExpression = require('../lib/searchOperators.js').searchNOTExpression;
-const searchPhrase = require('../lib/searchOperators.js').searchPhrase;
-const searchExactMatch = require('../lib/searchOperators.js').searchExactMatch;
+const searchORExpression = require('../lib/searchOperators').searchORExpression;
+const searchANDExpression = require('../lib/searchOperators').searchANDExpression;
+const searchNOTExpression = require('../lib/searchOperators').searchNOTExpression;
+const searchPhrase = require('../lib/searchOperators').searchPhrase;
```
Comment (netguru-hound) [inline test/searchOperators.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

File: test/searchOperators.js [hunk 31]
```diff
@@ -1,11 +1,11 @@
...
 
 const expect = require('chai').expect;
-const searchORExpression = require('../lib/searchOperators.js').searchORExpression;
-const searchANDExpression = require('../lib/searchOperators.js').searchANDExpression;
-const searchNOTExpression = require('../lib/searchOperators.js').searchNOTExpression;
-const searchPhrase = require('../lib/searchOperators.js').searchPhrase;
-const searchExactMatch = require('../lib/searchOperators.js').searchExactMatch;
+const searchORExpression = require('../lib/searchOperators').searchORExpression;
+const searchANDExpression = require('../lib/searchOperators').searchANDExpression;
+const searchNOTExpression = require('../lib/searchOperators').searchNOTExpression;
```
Comment (netguru-hound) [inline test/searchOperators.js]: Line is too long.<br>'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

File: test/searchOperators.js [hunk 32]
```diff
@@ -1,11 +1,11 @@
 'use strict';
 
 const expect = require('chai').expect;
-const searchORExpression = require('../lib/searchOperators.js').searchORExpression;
-const searchANDExpression = require('../lib/searchOperators.js').searchANDExpression;
-const searchNOTExpression = require('../lib/searchOperators.js').searchNOTExpression;
-const searchPhrase = require('../lib/searchOperators.js').searchPhrase;
-const searchExactMatch = require('../lib/searchOperators.js').searchExactMatch;
+const searchORExpression = require('../lib/searchOperators').searchORExpression;
+const searchANDExpression = require('../lib/searchOperators').searchANDExpression;
```
Comment (netguru-hound) [inline test/searchOperators.js]: Line is too long.<br>'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

File: test/searchOperators.js [hunk 33]
```diff
@@ -1,11 +1,11 @@
 'use strict';
 
 const expect = require('chai').expect;
-const searchORExpression = require('../lib/searchOperators.js').searchORExpression;
-const searchANDExpression = require('../lib/searchOperators.js').searchANDExpression;
-const searchNOTExpression = require('../lib/searchOperators.js').searchNOTExpression;
-const searchPhrase = require('../lib/searchOperators.js').searchPhrase;
-const searchExactMatch = require('../lib/searchOperators.js').searchExactMatch;
+const searchORExpression = require('../lib/searchOperators').searchORExpression;
```
Comment (netguru-hound) [inline test/searchOperators.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

File: test/mapFilter.js [hunk 34]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 const expect = require('chai').expect;
-const mapFilter = require('../lib/mapFilter.js');
+const mapFilter = require('../lib/mapFilter');
```
Comment (netguru-hound) [inline test/mapFilter.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

File: test/logicalExpressionSchema.js [hunk 35]
```diff
@@ -2,7 +2,7 @@
 
 const Joi = require('joi');
 const expect = require('chai').expect;
-const logicalExpressionSchema = require('../lib/logicalExpressionSchema.js');
+const logicalExpressionSchema = require('../lib/logicalExpressionSchema');
```
Comment (netguru-hound) [inline test/logicalExpressionSchema.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

File: lib/mapFilter.js [hunk 36]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-const filterSchema = require('./filterSchema.js');
+const filterSchema = require('./filterSchema');
```
Comment (netguru-hound) [inline lib/mapFilter.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'require' is not defined.

**Pull request merged**